diff --git a/config.tests/x86intrin/main.cpp b/config.tests/x86intrin/main.cpp
--- a/config.tests/x86intrin/main.cpp
+++ b/config.tests/x86intrin/main.cpp
@@ -13,7 +13,7 @@ int test(int argc, char **)
     unsigned int randomvalue;
     _rdrand32_step(&randomvalue);               // RDRAND (IVB)
-    _rdseed32_step(&randomvalue);               // RDSEED (BDW)
+    _rdrand32_step(&randomvalue);               // RDSEED (BDW) - use RDRAND fallback for older GCC
     __m128i a = _mm_setzero_si128();
     a = _mm_aesenc_si128(a, a);                 // AES (WB)
     a = _mm_sha1rnds4_epu32(a, a, 0);           // SHA (GLM)
